server/policy_test.go

```diff
@@ -151,3 +166,113 @@ func TestEvaluatePolicyFailure(t *testing.T) {
 		})
 	}
 }
+
+func TestSevSnpSignedUefi(t *testing.T) {
+	// Generate fake signed reference measurements.
+	fw := ovmfsev.CleanExample(t, 0x200000)
+	meas, err := gcesev.LaunchDigest(gcesev.LaunchOptionsDefault(), fw)
+	if err != nil {
+		t.Fatal(err)
+	}
+	certdir := t.TempDir()
+	keydir := t.TempDir()
+	outdir := t.TempDir()
+	outdir2 := t.TempDir()
+	if err := devkeys.DumpTo(&devkeys.Options{
+		KeyDir:   keydir,
+		CertRoot: certdir,
+		CertDir:  "signer_certs",
+		Bucket:   "certs-dev",
+	}); err != nil {
+		t.Fatal(err)
+	}
+	fwdir := t.TempDir()
+	fwPath := path.Join(fwdir, "fw.fd")
+	if err := os.WriteFile(fwPath, fw, 0666); err != nil {
+		t.Fatal(err)
+	}
+
+	nonprod.RootCmd.SetArgs([]string{"endorse",
+		"--uefi", fwPath,
+		"--key_dir", keydir,
+		"--bucket_root", certdir,
+		"--root_path", "root.crt",
+		"--add_snp",
+		"--out_root", outdir,
+		"--snapshot_dir", "snap",
+	})
+	if err := nonprod.RootCmd.Execute(); err != nil {
+		t.Fatal(err)
+	}
+	dents, err := os.ReadDir(outdir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	dents2, err := os.ReadDir(outdir2)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println("dents for", outdir, len(dents))
+	for _, dirent := range dents {
+		fmt.Println(dirent.Name())
+	}
+	fmt.Println("dents for", outdir2, len(dents2))
+	for _, dirent := range dents2 {
+		fmt.Println(dirent.Name())
+	}
+	rim, err := os.ReadFile(path.Join(outdir, "snap", "fw.fd.signed"))
+	if err != nil {
+		t.Fatalf("endorsement missing: %v", err)
+	}
+
+	// Generate fake VCEK certificate that can pass the basic validation checks.
+	block, _ := pem.Decode(devkeys.RootPEM)
+	pregenRoot, err := x509.ParsePKCS8PrivateKey(block.Bytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+	block, _ = pem.Decode(devkeys.RootCert)
+	rootDER := block.Bytes
+	insecureRandomness := rand.New(rand.NewSource(0xc0de))
+	vcek, err := ecdsa.GenerateKey(elliptic.P384(), insecureRandomness)
+	if err != nil {
+		t.Fatalf("could not generate fake VCEK: %v", err)
+	}
+	b := stest.AmdSignerBuilder{
+		Keys: &stest.AmdKeys{
+			Ark:  pregenRoot.(*rsa.PrivateKey),
+			Ask:  pregenRoot.(*rsa.PrivateKey),
+			Asvk: pregenRoot.(*rsa.PrivateKey),
+			Vcek: vcek,
+		},
+	}
+	signer, _ := b.TestOnlyCertChain()
+	vcekCert := signer.Vcek.Raw
+	ms := &pb.MachineState{
+		TeeAttestation: &pb.MachineState_SevSnpAttestation{
+			SevSnpAttestation: &spb.Attestation{
+				Report: &spb.Report{
+					Policy:      0x20000,
+					Measurement: meas,
+				},
+				CertificateChain: &spb.CertificateChain{
+					VcekCert: vcekCert,
+					Extras: map[string][]byte{
+						gcesev.GCEFwCertGUID: rim,
+					},
+				},
+			},
+		},
+	}
+
+	pol := &pb.Policy{
+		SevSnp: &pb.SevSnpPolicy{
+			Uefi: &pb.RIMPolicy{
+				RootCerts: [][]byte{rootDER},
+			},
+		},
+	}
+	if err := EvaluatePolicy(ms, pol); err != nil {
```

Comment:
Consider adding tests cases for nil policy + nil UEFI policy

Reply:
Thanks, found a bug.